ansible_base/lib/testing/fixtures.py

```diff
@@ -130,16 +130,19 @@ def expected_log(no_log_messages):
 
     @contextmanager
     def f(patch, severity, substr, assert_not_called=False):
-        with mock.patch(patch) as logger:
+        with mock.patch(f'{patch}.{severity}') as logger:
             with no_log_messages():
                 yield
-            sev_logger = getattr(logger, severity)
+
+            call_count = 0
+            for call in logger.call_args_list:
+                if substr in call.args[0]:
```

Comment:
Hmm, yeah that's a good catch, that if the logger is called with other messages in the same block, we'd fail. :+1: good fix